src/ctap/mod.rs

```diff
@@ -521,9 +543,11 @@ where
             None => return Err(Ctap2StatusCode::CTAP2_ERR_VENDOR_RESPONSE_CANNOT_WRITE_CBOR),
         };
         auth_data.extend(cose_key);
-        if use_hmac_extension {
-            let extensions = cbor_map! {
-                "hmac-secret" => true,
+        if has_extension_output {
+            let hmac_secret_output = if use_hmac_extension { Some(true) } else { None };
+            let extensions = cbor_map_options! {
+                "hmac-secret" => hmac_secret_output,
+                "credProtect" => cred_protect_policy.map(|policy| policy as i64),
```

Comment:
We should be able to remove this `map` if `From<Policy> for cbor::Value` is implemented.

Reply:
Removed.